src/script/common/c_content.cpp

```diff
@@ -526,6 +526,12 @@ ContentFeatures read_content_features(lua_State *L, int index)
 	// Amount of light the node emits
 	f.light_source = getintfield_default(L, index,
 			"light_source", f.light_source);
+	if (f.light_source > LIGHT_MAX) {
+		warningstream << "Node " << f.name.c_str()
+			<< " had greater light_source than 14, it was reduced."
+			<< std::endl;
+		f.light_source = LIGHT_MAX;
+	}
```

Comment:
Agreed, clearly that check did not help in the case of that server with the node causing the crash.  I think an additional sanity check here is well warranted.